<a href="https://colab.research.google.com/github/srishtiarora42006-hue/Answer/blob/main/Pandas_Dataframe.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Q1.** Creating a DataFrame

Create a Pandas DataFrame using a Python dictionary with the following columns:

Product

Region

Sales

Quantity


Add at least 5 rows of sample data and display the DataFrame.

In [11]:
Data = { "Product" : ["pencils","pen", "notebooks", "books", "covers"], "Region":["North", "south", "East", "West", "Central"], "Sales": [10000,15000,30000,20000,40000], "Quantity" : [2000,5000,1000,50,2000]}
pd.DataFrame(Data)

,Product,Region,Sales,Quantity
0,pencils,North,10000,2000
1,pen,south,15000,5000
2,notebooks,East,30000,1000
3,books,West,20000,50
4,covers,Central,40000,2000


In [12]:
import pandas as pd

**Q2.** Data Inspection

Perform the following inspection tasks:

Display the first 5 rows of the dataset.

Display the last 5 rows.

Show the structure of the dataset using info().

Display the shape of the dataset.

In [13]:
df = pd.read_excel("Retail_dataset (1).xlsx")

In [14]:
print(df.head())

   Order_ID Customer   Product Region  Sales  Quantity
0      1001    Sneha    Laptop   West   1056         2
1      1002     Neha    Tablet  South    904         7
2      1003    Meera    Laptop  South   1597         5
3      1004    Rohan  Keyboard  North    777         8
4      1005    Priya     Phone  South   2028         4


In [15]:
print(df.tail())

    Order_ID Customer   Product Region  Sales  Quantity
85      1086    Pooja    Laptop   East   1330         2
86      1087    Anita  Keyboard   West    741         3
87      1088    Karan   Monitor   East    875         2
88      1089      Raj    Tablet  North   2213         5
89      1090    Priya    Laptop  South   1270         2


In [16]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90 entries, 0 to 89
Data columns (total 6 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Order_ID  90 non-null     int64 
 1   Customer  90 non-null     object
 2   Product   90 non-null     object
 3   Region    90 non-null     object
 4   Sales     90 non-null     int64 
 5   Quantity  90 non-null     int64 
dtypes: int64(3), object(3)
memory usage: 4.3+ KB
None


In [17]:
print(df.shape)

(90, 6)


**Q3.** Column Selection

Select and display only the following columns:

Product

Region

Sales

In [18]:
df[["Product", "Region","Sales"]]

,Product,Region,Sales
0,Laptop,West,1056
1,Tablet,South,904
2,Laptop,South,1597
3,Keyboard,North,777
4,Phone,South,2028
...,...,...,...
85,Laptop,East,1330
86,Keyboard,West,741
87,Monitor,East,875
88,Tablet,North,2213


**Q4.** Indexing using .loc
Using .loc, perform the following:

Display the row with index 10.

Display the Sales value of the row with index 20




In [20]:
df.loc[10]

,10
Order_ID,1011
Customer,Raj
Product,Monitor
Region,North
Sales,1693
Quantity,6


In [22]:
df.loc[20,'Sales']

np.int64(1021)

**Q5.** Indexing using .iloc

Using .iloc, display:

The first 6 rows and first 4 columns.

The Sales value from the 3rd row.

In [23]:
df.iloc[:6,:4]

,Order_ID,Customer,Product,Region
0,1001,Sneha,Laptop,West
1,1002,Neha,Tablet,South
2,1003,Meera,Laptop,South
3,1004,Rohan,Keyboard,North
4,1005,Priya,Phone,South
5,1006,Rahul,Tablet,West


In [26]:
df.iloc[3,4]

np.int64(777)

**Q6.** Row Slicing

Display:

The first 10 rows of the dataset.

Rows 15 to 25.

In [27]:
df.iloc[0:10]

,Order_ID,Customer,Product,Region,Sales,Quantity
0,1001,Sneha,Laptop,West,1056,2
1,1002,Neha,Tablet,South,904,7
2,1003,Meera,Laptop,South,1597,5
3,1004,Rohan,Keyboard,North,777,8
4,1005,Priya,Phone,South,2028,4
5,1006,Rahul,Tablet,West,638,2
6,1007,Amit,Monitor,South,1004,7
7,1008,Sneha,Monitor,East,1269,8
8,1009,Priya,Phone,North,827,8
9,1010,Amit,Phone,West,1998,4


In [28]:
df.iloc[15:26]

,Order_ID,Customer,Product,Region,Sales,Quantity
15,1016,Karan,Monitor,West,796,3
16,1017,Neha,Laptop,East,2015,2
17,1018,Anita,Phone,North,1637,3
18,1019,Amit,Phone,East,748,7
19,1020,Neha,Laptop,West,1503,6
20,1021,Priya,Laptop,North,1021,4
21,1022,Raj,Tablet,South,886,7
22,1023,Rahul,Laptop,South,2267,3
23,1024,Karan,Monitor,East,814,7
24,1025,Meera,Monitor,West,1926,6


**Q7.** Filtering Data

Filter the dataset to display:

Rows where Region = "East".

Rows where Sales > 1500.

In [30]:
df[df["Region"]== "East"]

,Order_ID,Customer,Product,Region,Sales,Quantity
7,1008,Sneha,Monitor,East,1269,8
11,1012,Rahul,Laptop,East,1022,5
16,1017,Neha,Laptop,East,2015,2
18,1019,Amit,Phone,East,748,7
23,1024,Karan,Monitor,East,814,7
25,1026,Anita,Tablet,East,839,5
31,1032,Amit,Laptop,East,2226,6
34,1035,Amit,Tablet,East,2104,3
35,1036,Rohan,Phone,East,2136,8
36,1037,Raj,Phone,East,1650,4


In [31]:
df[df["Sales"]> 1500]

,Order_ID,Customer,Product,Region,Sales,Quantity
2,1003,Meera,Laptop,South,1597,5
4,1005,Priya,Phone,South,2028,4
9,1010,Amit,Phone,West,1998,4
10,1011,Raj,Monitor,North,1693,6
13,1014,Rohan,Phone,South,2078,3
16,1017,Neha,Laptop,East,2015,2
17,1018,Anita,Phone,North,1637,3
19,1020,Neha,Laptop,West,1503,6
22,1023,Rahul,Laptop,South,2267,3
24,1025,Meera,Monitor,West,1926,6


**Q8.** Multiple Condition Filtering

Display rows where:

Region is West

Sales is greater than 1200

In [36]:
df[(df["Region"]== "West")&( df["Sales"] > 1200)]

,Order_ID,Customer,Product,Region,Sales,Quantity
9,1010,Amit,Phone,West,1998,4
19,1020,Neha,Laptop,West,1503,6
24,1025,Meera,Monitor,West,1926,6
29,1030,Rohan,Monitor,West,1932,5
42,1043,Meera,Keyboard,West,1515,6
43,1044,Vikas,Laptop,West,2248,5
49,1050,Karan,Monitor,West,2034,7
55,1056,Rahul,Laptop,West,1481,1
58,1059,Karan,Tablet,West,2404,6
59,1060,Karan,Tablet,West,1651,2


**Q9.** Descriptive Statistics

Generate summary statistics using:

describe()

From the result identify:

Mean sales value

Maximum quantity sold

Minimum sales value

In [37]:
df.describe()

,Order_ID,Sales,Quantity
count,90.000000,90.000000,90.000000
mean,1045.500000,1406.888889,4.700000
std,26.124701,531.748302,2.179836
min,1001.000000,515.000000,1.000000
25%,1023.250000,890.500000,3.000000
50%,1045.500000,1402.000000,5.000000
75%,1067.750000,1857.250000,7.000000
max,1090.000000,2455.000000,8.000000


Mean sales value -> 1406.89

Maximum quantity sold -> 8 units

Minimum sales value -> ₹515

**Q10.** Aggregation Analysis

Using the Sales column, calculate:

Total sales

Average sales

Maximum sales

Minimum sales

In [38]:
total_sales = df['Sales'].sum()
print(total_sales)

126620


In [39]:
Average_sales = df['Sales'].mean()
print(Average_sales)

1406.888888888889


In [40]:
Maximum_sales = df['Sales'].max()
print(Maximum_sales)

2455


In [41]:
Minimum_sales = df['Sales'].min()
print(Minimum_sales)

515
